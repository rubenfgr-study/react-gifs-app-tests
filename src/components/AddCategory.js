import pt from "prop-types";
import { useState } from "react";

export const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState("");

  const handleInputChange = (e) => {
    setInputValue(e.target.value);

    console.log("handleInputchange launch");
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    if (inputValue.trim().length > 2) {
      setCategories((categories) => [inputValue, ...categories]);
      setInputValue("");
    }
  };

  return (
    <form onSubmit={handleSubmit}>
      <h3>{inputValue}</h3>
      <input type="text" value={inputValue} onChange={handleInputChange} />
    </form>
  );
};

AddCategory.propTypes = {
  setCategories: pt.func.isRequired,
};
