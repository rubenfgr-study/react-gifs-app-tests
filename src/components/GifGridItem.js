import "animate.css";
import propTypes from "prop-types";
import React from "react";

export const GifGridItem = ({ title, url }) => {
  return (
    <div className="card animate__animated animate__fadeInRight">
      <img src={url} alt={title} />
      <p>{title}</p>
    </div>
  );
};

GifGridItem.prototypes = {
  title: propTypes.string.isRequired,
  url: propTypes.string.isRequired,
};
