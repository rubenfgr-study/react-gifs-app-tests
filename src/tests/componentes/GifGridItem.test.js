import { shallow } from "enzyme";
import { GifGridItem } from "../../components/GifGridItem";

describe("GifGridItem Test", () => {
  const title = "Title";
  const url = "http://localhost";
  const wrapper = shallow(<GifGridItem title={title} url={url} />);

  test("GifGridItem test snapshoot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("debe de tener un parrafo con el title", () => {
    const p = wrapper.find("p");
    expect(p.text().trim()).toBe(title);
  });

  test("debe tener una imagen  con el src = url y alt = title", () => {
    const img = wrapper.find("img");
    expect(img.prop("src")).toBe(url);
    expect(img.prop("alt")).toBe(title);
  });

  test("debe tener animate_fadeIn", () => {
    const div = wrapper.find("div");
    expect(div.hasClass("animate__animated")).toBe(true);
  });
});
