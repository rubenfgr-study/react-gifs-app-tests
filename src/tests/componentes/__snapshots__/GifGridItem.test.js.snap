// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGridItem Test GifGridItem test snapshoot 1`] = `
<div
  className="card animate__animated animate__fadeInRight"
>
  <img
    alt="Title"
    src="http://localhost"
  />
  <p>
    Title
  </p>
</div>
`;
